### DISCUSSION TITLE
document what wayland protocols have been implemented

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#232] [General]
e.g. doing checkboxes to track progress, say going by https://wayland.app/protocols/

### stable
- Presentation time
- Viewporter
- XDG shell
- Linux DMA-BUF
### staging
- XDG activation
- DRM lease
- Session lock
- Single-pixel buffer
- Content type hint
- Idle notify
- Tearing control
- Xwayland shell
- Fractional scale
- Cursor shape
- Foreign toplevel list
- Security context
- Transient seat
- XDG toplevel drag
### unstable
- Fullscreen shell
- Idle inhibit
- Input method v1
- Input timestamps
- Keyboard shortcuts inhibit
- Linux explicit synchronization (dma-fence)
- Pointer constraints
- Pointer gestures
- Primary selection
- Relative pointer
- Tablet
- Text input
- XDG decoration
- XDG foreign
- XDG output
- XWayland keyboard grabbing
### wlr
- wlr data control
- wlr export DMA-BUF
- wlr foreign toplevel management
- wlr gamma control
- wlr input inhibitor
- wlr layer shell
- wlr output management
- wlr output power management
- wlr screencopy
- wlr virtual pointer

**Comment by YaLTeR** [comment]
I'm not sure a list like this is entirely useful for a compositor (as opposed to a library like Smithay). As a user, you care about specific functionality like fractional scaling; this doesn't always cleanly map to a particular protocol. Also, some protocols don't make all that much of a difference, whether in general or for niri in particular.

**Reply by KiaraGrouwstra** [reply]
hm, as per the open issues, it looks like some would need handling over here as well rather than just by smithy right?
i see your point about users wanting to know what functionality they can use.
i guess what a list like this might help answer tho (probably with the names more like `keyboard-shortcuts-inhibit-unstable-v1` rather than how i put them here) is the question of: if i'm using this wm with [insert bar module / whatever], will it work? i guess the protocol names make for a clear formal way to help communicate stuff like that.
but yeah, i guess explanations (if not examples of compatible applications) might help answer other relevant questions.

**Reply by YaLTeR** [reply]
That makes sense, thinking about it from the perspective of what apps people might want to use, or what specific features like keyboard shortcuts inhibiting. Guess I can make a list whenever I'm not lazy lol

**Comment by 0xDEAD-C0DE** [comment]
@KiaraGrouwstra has some valid argument here. For example, I was trying to use wl-sunset and gammset and could not use neither one due to the required wlr-gamma-control-unstable-v1 protocol.

**Comment by kchibisov** [comment]
You can get all the protocols and their versions within a oneliner like

```
WAYLAND_DEBUG=1 alacritty 2>&1 | rg  -i 'wl_registry.*global.*"(.*)", ([0-9]+)' -o -r "\$1@v\$2" | sort | uniq
```

you can use any app instead of alacritty, though, not flatpak one.

**Reply by YaLTeR** [reply]
You can also just run `wayland-info` 😄

**Reply by kchibisov** [reply]
I mean it requires installing weston stuff iirc. I have no clue where it's coming from in the first place and I never had it on any of my systems.

**Reply by sodiboo** [reply]
> I mean it requires installing weston stuff iirc.

no? from what i can tell, the package with this tool is called `wayland-utils` on pretty much all distros. it's not directly related to weston (although maybe it was historically? i don't know)

**Reply by YaLTeR** [reply]
It used to be weston-info, but it's been renamed to wayland-info a while ago.

**Reply by FreeFull** [reply]
There's also a GUI program named "waycheck" that lists all the supported wayland protocols in a table.

**Reply by YaLTeR** [reply]
Oh, that's pretty handy!
